dotnet-desktop-guide/net/wpf/documents/snippets/how-to-display-print-dialog/csharp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Xps.Packaging;
+using Microsoft.Win32;
+
+namespace CodeSampleCsharp
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            // Events
+            btnBrowse.Click += BtnBrowse_Click;
+            btnPrint.Click += BtnPrint_Click;
+        }
+
+        private void BtnBrowse_Click(object sender, RoutedEventArgs e)
+        {
+            // Configure an open file dialog box
+            OpenFileDialog openFileDialog = new()
+            {
+                FileName = "Document", // Default file name
+                DefaultExt = ".xps", // Default file extension
+                Filter = "Text documents (.xps)|*.xps" // Filter files by extension
+            };
+
+            // Show open file dialog box.
+            bool? result = openFileDialog.ShowDialog();
+
+            // Process open file dialog box results
+            if (result == true)
+            {
+                // Show selected XPS file path
+                txtXpsFilePath.Text = openFileDialog.FileName;
+            }
+        }
+
+        private void BtnPrint_Click(object sender, RoutedEventArgs e)
+        {
+            // Check the file exists
+            if (!File.Exists(txtXpsFilePath.Text))
+            {
+                _ = MessageBox.Show("First select an XPS file.");
```

Comment:
We don't use discards on unused return values. Discards are useful for when you have multiple return values and you want to discard one of them, or you're discarding an output parameter.
```suggestion
                MessageBox.Show("First select an XPS file.");
```